keps/sig-network/1111-dynamically-re-configure-lb-node-set/README.md

```diff
@@ -0,0 +1,681 @@
+  implement the feature. The HCs will fail on all cluster nodes for ETP=cluster
+  services and ingress will be broken.
+
+#### Mitigations
+
+- Don't enable the feature flag if you are not 100 % sure your service proxy
+  supports the new functionality.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Prerequisite testing updates
+
+#### Unit tests
+
+Components in the architecture presented above are already existing (and
+supposedly are already unit tested): kube-proxy's healthz server, kube-proxy's
+"ETP=local" server. We would need to add tests which validate the change in
+behaviour for opening up additional ports for ETP=cluster services and validate
+that the difference in HC logic between ETP services is captured correctly.
+
+#### Integration tests
+
+Kube-proxy is mostly tested via unit tests and e2e, not integration, and this is
+not expected to change.
+
+#### e2e tests
+
+Kubernetes in general needs to extended its load balancing test suite with
+disruption tests, this might be the right effort we need to get that ball
+rolling. Testing would include: validation that HC probe's return a respond
+corresponding to the "state of the Kubernetes cluster world" during different
+disruption scenarios.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature flag for changing the API for ETP=cluster services and
+  include `healthCheckNodePort` for them. This will help testing and graduation
+  to beta and stable.  
+- Changes to data plane components get implemented (Kube-proxy)
+- Changes to control plane components get implemented, but depends on the
+  feature flag being enabled or not.
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Documentation is changed for ETP=cluster services to highlight the change with
+  regards to `healthCheckNodePort`.
+- Testing of Kubernetes with the feature flag enabled is green and reports no
+  major incidents.
+
+#### GA
+
+- Version skew delta has passed.
+
+### Upgrade / Downgrade Strategy
```

Comment:
I have service class A and I upgrade, I will not have the HCNodePort
I have a service classA with HCNodePort and I downgrade, will validation fail this Service?

Reply:
> I have service class A and I upgrade, I will not have the HCNodePort

The idea is that it the class A service gets assigned an HCNP upon upgrading, automatically just like class B services get assigned one automatically today. 

> I have a service classA with HCNodePort and I downgrade, will validation fail this Service?

No, I would expect the field to be unset, not invalidated.

Reply:
you should mention these things in this section :)

Reply:
good example of upgrades problems https://github.com/kubernetes/kubernetes/issues/112793 , this one looks hard to spot though